File: IdeaProjects/MessageReader/src/main/java/messageReader/monitor/StatusMessageMonitor.java
package messageReader.monitor;


import messageReader.Notification;
import messageReader.NotificationPublisher;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.Map;

import static messageReader.MessageReaderConstants.ENTITY_NAME;
import static messageReader.MessageReaderConstants.STATUS;

public class StatusMessageMonitor  {

    private Logger logger = LoggerFactory.getLogger(StatusMessageMonitor.class);

    @Autowired
    public StatusMonitorStrategy statusMonitorStrategy;

    @Autowired
    public NotificationPublisher notificationPublisher;

    public void monitor(Map<String, String> message) {
        logger.debug(message.get(ENTITY_NAME) + " being monitored in the Status Monitor");

        if (message.containsKey(STATUS) && statusMonitorStrategy.isStatusInError(message)) {
            logger.info("-- CREATING NOTIFICATION: " + message.get(ENTITY_NAME) + " -- ERROR --");
            notificationPublisher.publish(new Notification(message));
        } else if (message.containsKey(STATUS) && !statusMonitorStrategy.isStatusInError(message)) {
            logger.info("-- CREATING NOTIFICATION: " + message.get(ENTITY_NAME) + " -- ACTIVE --");
            notificationPublisher.publishResolve(new Notification(message));
        } else {
            logger.error("--- Invalid Message Type Hit Monitor ---");
        }
    }
}

File: IdeaProjects/ScalaTutorial/ScalaMessageReader/src/main/scala/scalaMessageReader/monitor/StatusMessageMonitor.scala
package scalaMessageReader.monitor

import org.slf4j.{Logger, LoggerFactory}
import org.springframework.beans.factory.annotation.Autowired

import scala.collection.mutable
import scalaMessageReader.{Notification, NotificationPublisher}
import scalaMessageReader.MessageReaderConstants.{ENTITY_NAME,STATUS}

class StatusMessageMonitor {

  @Autowired val monitorStrategy : StatusMonitorStrategy = null
  @Autowired val notificationPublisher : NotificationPublisher = null

  val logger : Logger = LoggerFactory.getLogger(classOf[StatusMessageMonitor])

  def monitor (message : mutable.Map[String,String]): Unit = {
    logger.debug(message(ENTITY_NAME) + " being monitored in the Status Monitor")

    if (message.contains(STATUS) && monitorStrategy.isInError(message)) {
      logger.info("-- CREATING NOTIFICATION: " + message(ENTITY_NAME) + " -- ERROR --")
      notificationPublisher.publish(new Notification(message))
    } else if (message.contains(STATUS) && !monitorStrategy.isInError(message)) {
      logger.info("-- CREATING NOTIFICATION: " + message(ENTITY_NAME) + " -- ACTIVE --")
      notificationPublisher.publishResolve(new Notification(message))
    } else logger.info("--- Invalid Message Type Hit Monitor ---")
  }

}
